src/background/tab-state.js

```diff
@@ -171,9 +190,19 @@ export default function TabState(initialState, onchange) {
 
     const debouncedFetch = new Promise((resolve, reject) => {
       const timerId = setTimeout(async () => {
-        let count;
+        let count = annotationCountCache.get(url);
+        if (count !== undefined) {
+          resolve(count);
+          return;
+        }
+
         try {
-          count = await uriInfo.getAnnotationCount(tabUrl);
+          count = await uriInfo.fetchAnnotationCount(url);
+          annotationCountCache.set(url, count);
+          setTimeout(
+            () => annotationCountCache.delete(url),
+            EXPIRATION_CACHE_MS
```

Comment:
I tried to think of scenarios where we could end up removing the cache entry at the wrong time or have multiple timeouts for the same URL, but I think this is OK because there can only be one pending request at once.

Reply:
It can only be one request pending _per tab_ at at time. There could be multiple request for the same URL for _different  tabs_.

In both cases, I can't think of any issues regarding removing the cache entry: it will always be `x`ms after the last successful request.